src/com/amazon/ionschema/internal/SchemaImpl.kt

```diff
@@ -154,21 +165,49 @@ internal class SchemaImpl private constructor(
 
                     val typeName = (it["type"] as? IonSymbol)?.stringValue()
                     if (typeName != null) {
-                        var newType = importedSchema.getType(typeName)
-                            ?: throw InvalidSchemaException(
-                                "Schema $childImportId doesn't contain a type named '$typeName'"
-                            )
+                        var importedType = importedSchema.getDeclaredType(typeName)
+
+                        if (importedType == null && allowTransitiveImports) {
+                            importedType = importedSchema.getType(typeName)?.also { type ->
+                                schemaSystem.logger(Warn) {
+                                    "Import is resolved transitively; $typeName is actually declared in " +
```

Comment:
Instead of saying `Import is resolved transitively` I feel if it is something similar to the option `ALLOW_TRANSITIVE_IMPORTS` then its easier to understand.

```kotlin
"Schema system allows transitive imports; $typeName is actually declared in schema '${type.schemaId}' -- $it' "
```

Reply:
In principle, I like this idea, but think the focus of the message should be on the problem that this particular type is imported in a way that is non-compliant with the spec rather than focusing on the fact that the schema system has transitive imports enabled.

What if there was a warning "code" like `WARN_TRANSITIVE_IMPORT` that for all of the logs related to this? It would be very easily searchable because it's all one word (w.r.t. search engines).

Eg.

```
"WARN_TRANSITIVE_IMPORT; $typeName is actually declared in schema '${type.schemaId}' -- $it' "
```